src/lib/bot/mod.rs

```diff
@@ -293,21 +376,42 @@ impl Bot {
         }
     }
 
-    pub fn get_streamelements_api(&mut self) -> &StreamElementsAPI {
+    pub fn get_streamelements_api(&self) -> &StreamElementsAPI {
         &self.api
     }
-    pub fn get_writer(&mut self) -> &Writer {
+    pub fn get_streamelements_api_mut(&mut self) -> &mut StreamElementsAPI {
+        &mut self.api
+    }
+    pub fn get_youtube_api(&self) -> &Option<YouTubePlaylistAPI> {
+        &self.yt_api
+    }
+    pub fn get_youtube_api_mut(&mut self) -> Option<&mut YouTubePlaylistAPI> {
+        self.yt_api.as_mut()
+    }
+    pub fn get_writer(&self) -> &Writer {
         &self.writer
     }
-    pub fn get_config(&mut self) -> &BotConfig {
+    pub fn get_writer_mut(&mut self) -> &mut Writer {
+        &mut self.writer
+    }
```

Comment:
You should make these fields public. A mutable reference allows one to `std::mem::swap()` in whatever nonsense they wish, so having a set of getters doesn't help.